pkg/networkservice/chains/usecase/usecase_test.go

```diff
@@ -520,9 +467,44 @@ func TestNSMGR_PassThroughLocal(t *testing.T) {
 	// Path length to first endpoint is 5
 	// Path length from NSE client to other local endpoint is 5
 	require.Equal(t, 5*(nsesCount-1)+5, len(conn.Path.PathSegments))
+
+	// Close
+	e, err := nsc.Close(ctx, conn)
+	require.NoError(t, err)
+	require.NotNil(t, e)
+
+	// Endpoint unregister
+	for i := 0; i < len(nsesReg); i++ {
+		_, err = s.domain.Nodes[0].EndpointRegistryClient.Unregister(ctx, nsesReg[i])
+		require.NoError(t, err)
+	}
 }
 
-func TestNSMGR_ShouldCorrectlyAddForwardersWithSameNames(t *testing.T) {
+func (s *nsmgrSuite) Test_ShouldCleanAllClientAndEndpointGoroutines() {
+	t := s.T()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	// We have lazy initialization in some chain elements in both networkservice, registry chains. So registering an
+	// endpoint and requesting it from client can result in new endless NSMgr goroutines.
+	testNSEAndClient(ctx, t, s.domain, &registry.NetworkServiceEndpoint{
+		Name:                "endpoint-init",
+		NetworkServiceNames: []string{"service-init"},
+	})
+
```

Comment:
This test was broken by https://github.com/networkservicemesh/sdk/pull/806/commits/725e8dbed932349dcf0d4566b67b4f52f5bde83d.
Please add here back check for the goroutines leaks:
```go
defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
```

Reply:
This is not related to this PR.